pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java

```diff
@@ -66,6 +65,7 @@ public ClientBuilder loadConf(Map<String, Object> config) {
     @Override
     public ClientBuilder serviceUrl(String serviceUrl) {
         conf.setServiceUrl(serviceUrl);
+        enableTls(serviceUrl.startsWith("pulsar+ssl") || serviceUrl.startsWith("https"));
```

Comment:
To retain previous behavior as well (even if it doesn't make a lot of practical sense), here we could do: 

```java
if (!conf.isUseTls()) {
   enableTls(serviceUrl.startsWith("pulsar+ssl") || serviceUrl.startsWith("https"));
}
```

Reply:
Sure, but this only works in one direction. If someone specifies enableTls(false) it will still be enabled if they use the URL types. Anyhow, changed.